jdb/assignment2/tree_no_parent.py

```diff
@@ -0,0 +1,97 @@
+import unittest
+from typing import Generator, List, Tuple, TypeVar
+
+T = TypeVar('T')
+TAncestors = Tuple[T]
+_StackOfT = List[T]
+
+class Node:
+    def __init__(self, value: T, left = None, right = None):
+        self.left = Node(*left) if left else None
+        self.value = value
+        self.right = Node(*right) if right else None
+        
+    def _ancestry_lines(self, stack: _StackOfT = None) -> Generator[_StackOfT, None, None]:
```

Comment:
The name of this function seems a little bit confusing. As I could understand from code, it starts from the root node and returns the lines of descendants, which are using in the function find_ancestors then. Am I understand correctly?

Reply:
This functions yields all possible ancestry lines in the tree. I think you got it right.

Since there is no parent pointer, I need to construct the ancestry lines by remembering all the ancestor of the current node I visit in a stack.